EmailResponder/FeedbackDecryptor/redactor.py

```diff
@@ -0,0 +1,356 @@
+# Copyright (c) 2020, Psiphon Inc.
+# All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import sys
+import types
+import re
+import json
+
+import logger
+import utils
+
+
+def redact_sensitive_values(obj):
+    '''
+    Redacts any sensitive values present in the given diagnostic information.
+    The leaves of the dictionary are searched and any sensitive values found
+    are redacted by modifying the dictionary directly.
+    '''
+
+    if isinstance(obj, utils.string_types) or not isinstance(obj, types.DictType):
+        return
+
+    try:
+        client_platform = obj["Metadata"]["platform"]
+        sys_info = obj["DiagnosticInfo"]["SystemInformation"]
+        client_version = sys_info["PsiphonInfo"]["CLIENT_VERSION"]
+    except KeyError:
+        return
+    
+    if isinstance(client_version, utils.string_types):
```

Comment:
This isn't new, but... I feel that this check should instead be:
```python
if not isinstance(client_version, int):
```
Etc. That way we're guaranteed that client_version is an int below (or we're not proceeding). With the code the way it is now, it could be a bool, float, dict, list, ...